### PR TITLE
Fix neo4j java home

### LINKED ISSUE #130106: Neo4j: does not start because JAVA_HOME not set
**Describe the bug**

Neo4j does not start because it cannot locate a Java Runtime.

**To Reproduce**

On my system, I installed with:
* `nix-env -iQ nixpkgs.jdk11`
* `nix-env -iQ nixpkgs.neo4j`

Then, running `neo4j version` shows: "The operation couldn’t be completed. Unable to locate a Java Runtime."

**Expected behavior**

Neo4j should start.

**Additional context**

Note that  `cat $HOME/.nix-profile/bin/neo4j` shows

```
#! /nix/store/pcjan45rssdn01cxx3sjg70avjg6c3ni-bash-4.4-p23/bin/bash -e
export PATH='/nix/store/ibssm80x933r060h8ms5g3fa2jv8j6vi-zulu8.54.0.21-ca-jdk-8.0.292/bin:/nix/store/7nharq38bpckrj87nbgqbi3na82jp721-which-2.21/bin:/nix/store/8kpxw5na07ggdl2bs8kiwysif7120r6g-gawk-5.1.0/bin'${PATH:+':'}$PATH
export JAVA_HOME=''
exec "/nix/store/lrsrks8z0snvvbxqcb6avsvrqg001w01-neo4j-3.5.14/share/neo4j/bin/neo4j"  "$@"
```

Note the line that sets `JAVA_HOME` to `''`.

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
- system: `"x86_64-darwin"`
 - host os: `Darwin 20.5.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.12`
 - channels(xpe): `"nixpkgs-21.11pre294471.51bb9f3e9ab"`
 - nixpkgs: `/Users/xpe/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: neo4j
```

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Closes #130106 

Smaller closure size.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130114` at 7f71b069 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>neo4j</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/servers/nosql/neo4j/default.nix:33:5:
```
   |
33 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/servers/nosql/neo4j/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19911,7 +19911,7 @@ in
 
   check_systemd = callPackage ../servers/monitoring/nagios/plugins/check_systemd.nix { };
 
-  neo4j = callPackage ../servers/nosql/neo4j { };
+  neo4j = callPackage ../servers/nosql/neo4j { jre = jre8_headless; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:19914]:
```suggestion
  neo4j = callPackage ../servers/nosql/neo4j {
    jre = jre8_headless;
  };
```

**Comment by Artturin** [comment]
https://git-rebase.io/#fixup

**Comment by roberth** [comment]
> apply the suggestions by r-rmcgibbo

It's rude to address someone in the imperative mood.

> https://git-rebase.io/#fixup

Probably meant to be helpful, but it is not. Please provide context when you post a link. What am I supposed to infer? That I need to fixup or squash something? Like what? I've only made independent changes. One refactoring and one real change. Those shouldn't be combined into a single commit.

I was trying to help someone who has a _real_ problem and all I get is this superficial, bureaucratic "feedback" that has no bearing on reality. Perhaps I should have just merged this PR myself. I'm sorry for wasting your time.

I hope you do allow incremental improvements from nixpkgs non-members.

**Comment by Artturin** [comment]
> > apply the suggestions by r-rmcgibbo
> 
> It's rude to address someone in the imperative mood.
> 
> > https://git-rebase.io/#fixup
> 
> Probably meant to be helpful, but it is not. Please provide context when you post a link. What am I supposed to infer? That I need to fixup or squash something? Like what? I've only made independent changes. One refactoring and one real change. Those shouldn't be combined into a single commit.
> 
> I was trying to help someone who has a _real_ problem and all I get is this superficial, bureaucratic "feedback" that has no bearing on reality. Perhaps I should have just merged this PR myself. I'm sorry for wasting your time.
> 
> I hope you do allow incremental improvements from nixpkgs non-members.

I didn't mean to be imperative

Sorry for not elaborating that I thought this could have been in the previous commit https://github.com/NixOS/nixpkgs/pull/130114/commits/2ed8c5f4a901c80e0cbf8f930f37022a02be608a

I will make better comments in the future

**Pull request merged**